packages/source-iottwinmaker/src/metadata-module/TwinMakerMetadataModule.ts

```diff
@@ -1,48 +1,39 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { GetEntityCommand, IoTTwinMakerClient, ListEntitiesCommand } from '@aws-sdk/client-iottwinmaker';
 import { isDefined } from '../time-series-data/utils/values';
-import { TwinMakerMetadataCache } from './TwinMakerMetadataCache';
 import type { EntitySummary, GetEntityResponse, ListEntitiesResponse } from '@aws-sdk/client-iottwinmaker';
 import type { ErrorDetails } from '@iot-app-kit/core';
+import { QueryClient } from '@tanstack/query-core';
 
 export class TwinMakerMetadataModule {
   private readonly workspaceId: string;
   private readonly tmClient: IoTTwinMakerClient;
-  private readonly cache: TwinMakerMetadataCache;
-
-  private readonly entityRequests: Record<string, Promise<GetEntityResponse>> = {};
-  private readonly listEntitiesByComponentTypeIdRequests: Record<string, Promise<EntitySummary[]>> = {};
+  private readonly cachedQueryClient: QueryClient;
 
   constructor(
     workspaceId: string,
     client: IoTTwinMakerClient,
-    cache: TwinMakerMetadataCache = new TwinMakerMetadataCache()
+    cachedQueryClient: QueryClient = new QueryClient({
+      defaultOptions: {
+        queries: {
+          staleTime: Infinity,
```

Comment:
If you set this to infinity that means data fetched from the server is never considered as stale. So If the data does change on the server, your application will not be aware of the changes until `queryClient.invalidateQueries()` is called.